From 0fc12ba75e2a3f1ef9d34ac475820082a61fcbaf Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 13 Jul 2023 20:29:43 -0500
Subject: [PATCH 3/3] net: ieee802154: wpanusb: Call _xmit_hw_error() upon
 transmission error

ieee802154_xmit_hw_error() is the right helper to call when a transmission
has failed for a non-determined (and probably not IEEE802.15.4 specific)
reason. Let's use this helper instead of open-coding it.

Based on: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?h=v6.5-rc1&id=35f34ee102a5b2376f4b4700ac017b8f2770e754

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/net/ieee802154/wpanusb.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ieee802154/wpanusb.c b/drivers/net/ieee802154/wpanusb.c
index 51eb0d32b291..b9af234474f3 100644
--- a/drivers/net/ieee802154/wpanusb.c
+++ b/drivers/net/ieee802154/wpanusb.c
@@ -148,9 +148,7 @@ static void wpanusb_tx_done(struct wpanusb *wpanusb, uint8_t seq)
 	} else {
 		dev_dbg(&udev->dev, "unknown ack %u\n", seq);
 
-		ieee802154_wake_queue(wpanusb->hw);
-		if (wpanusb->tx_skb)
-			dev_kfree_skb_irq(wpanusb->tx_skb);
+		ieee802154_xmit_hw_error(wpanusb->hw, wpanusb->tx_skb);
 	}
 }
 
-- 
2.39.2

